src/quantities.jl

```diff
@@ -265,17 +265,17 @@ isapprox(x::Number, y::AbstractQuantity; kwargs...) = isapprox(y, x; kwargs...)
 function isapprox(
     x::AbstractArray{<:AbstractQuantity{T1,D,U1}},
     y::AbstractArray{<:AbstractQuantity{T2,D,U2}};
-    rtol::Real=Base.rtoldefault(T1,T2,0),
     atol=zero(Quantity{real(T1),D,U1}),
+    rtol::Real=Base.rtoldefault(T1,T2,atol>zero(atol)),
+    nans::Bool=false,
     norm::Function=norm,
 ) where {T1,D,U1,T2,U2}
-
     d = norm(x - y)
     if isfinite(d)
-        return d <= atol + rtol*max(norm(x), norm(y))
+        return iszero(rtol) ? d <= atol : d <= max(atol, rtol*max(norm(x), norm(y)))
     else
         # Fall back to a component-wise approximate comparison
-        return all(ab -> isapprox(ab[1], ab[2]; rtol=rtol, atol=atol), zip(x, y))
+        return mapreduce((a, b) -> isapprox(a, b; rtol=rtol, atol=atol, nans=nans), &, x, y)
```

Comment:
@sostock, checking the documentation, I noticed this:
```
!!! compat "Julia 1.2" `mapreduce` with multiple iterators requires Julia 1.2 or later.
```

Reply:
Maybe the solution to this is to use a `map` followed by a `reduce`:
```
reduce(&, map((a, b) -> isapprox(a, b; rtol=rtol, atol=atol, nans=nans), x, y))
```
What do you think @sostock?

Reply:
> Maybe the solution to this is to use a map followed by a reduce

Since that allocates a new array, I wouldn’t do it. My suggestion would be to revert to the old version for now (just with the added `nans` keyword argument):
```suggestion
        return all(ab -> isapprox(ab[1], ab[2]; rtol=rtol, atol=atol, nans=nans), zip(x, y))
```

But at some point we should just remove support for Julia <1.6, then we can change it to the `mapreduce` version.

Reply:
@sostock done